### ISSUE TITLE
Search instances in Survey

### ISSUE THREAD

**Issue by elmps2018** [opendatakit/tool-suite-X#225]
On the first screen in survey, listing the instances, it would be great to be able to search for an instance! This functionality is not available in survey but is available in Tables: https://forum.odk-x.org/t/how-to-search-a-form-by-an-id/463
Also/related: https://github.com/odk-x/tool-suite-X/issues/110

**Comment by khushishikhu** [comment]
@elmps2018 I am an outreachy participant. May I work on this issue?

**Comment by elmps2018** [comment]
Hi @khushishikhu! Yes, I have assigned it to you! Let us know if you have questions!

**Comment by khushishikhu** [comment]
@elmps2018 I want to ask that the search instance to be added in Survey should be by ID only as it is available in the Tables ?

**Comment by elmps2018** [comment]
Great question! Since it is long/hard to search the uuid, it might work better to search the instance name that is displayed (the settings sheet of a form sets up what the instance name is, if someone wants something different than the uuid).

**Comment by khushishikhu** [comment]
![Screenshot_2021-03-31-12-35-40-74.jpg](https://user-images.githubusercontent.com/65439761/113104334-083fb880-921e-11eb-841a-db413a3b307a.jpg)

@elmps2018 I am actually a newbie with android can you please help me with what type of dependencies is missing while I was running the ODK-X Survey in my device.

**Comment by wbrunette** [comment]
@khushishikhu in general it's good to check out our docs website.

Here are the basic steps for installing needed depedendencies: https://docs.odk-x.org/basics-install/

**Comment by khushishikhu** [comment]
@wbrunette I have installed the dependencies( .apk files of services and OI file manager) given in the document in my device. But still this same error is coming up. please help me to resolve this

**Comment by linl33** [comment]
@khushishikhu Do you happen to be on Android 11? If you're on Android 11, you would need to either use an older version or build the apps from source (because of https://github.com/odk-x/tool-suite-X/issues/193).

If you'd like to build from source, see the README on https://github.com/odk-x/services and https://github.com/odk-x/survey

**Comment by khushishikhu** [comment]
Thank you @linl33 and @wbrunette  I have successfully set up the app on my device. @elmps2018 I want to ask that do I have to make the search instance to every form we open in ODK-X Survey?

**Comment by linl33** [comment]
@khushishikhu The screen that displays the list of instances is shared across all forms (unless otherwise customized). The code for the screen is at https://github.com/odk-x/app-designer/blob/development/app/system/survey/templates/instances.handlebars and https://github.com/odk-x/app-designer/blob/f5fdafa20f076fd04513e548dd63446071f6ad03/app/system/survey/js/prompts.js#L570

In the future we might want to have a setting to turn search on/off but don't worry about that for now.

**Comment by khushishikhu** [comment]
@linl33 Thank you so much for providing me this , it's really of great help for me.  
I have a doubt that is this app-designer works for both odk-x survery and odk-x Tables?

**Comment by khushishikhu** [comment]
![image](https://user-images.githubusercontent.com/65439761/113394400-cc3f5b80-93b5-11eb-9fe2-6177f9faeb51.png)
@linl33 I have structured the Search box in household instanced( tried to apply in one form) Please let me know if anything needs to be refined here?

**Comment by wbrunette** [comment]
@khushishikhu you should test it out by creating several instances with several different forms. Once you are ready please open a Pull Request for a code review.

**Comment by wbrunette** [comment]
@khushishikhu once you finish this issue you might be interested in this issue since they are related https://github.com/odk-x/tool-suite-X/issues/110

**Comment by khushishikhu** [comment]
@wbrunette yes sure, Even I was thinking to work on that after resolving this issue as they are related but someone was already engaged on that issue.

**Comment by wbrunette** [comment]
@khushishikhu #110 is now available

**Comment by khushishikhu** [comment]
@wbrunette I have researched a lot and written the code for search , but unfortunately it fails to work. I need some help here to implement that, as I am not getting what else is currently missing in my code. That it doesn't work, Please can you give me some reference to read for how I can move further to solve this issue, it would be really helpful for me

**Comment by wbrunette** [comment]
@linl33 is more familiar with this code than I am. Let's see what he suggestions.

**Comment by linl33** [comment]
@khushishikhu The issue with your implementation is that you tried to take a sample for customizing ODK-X Tables to use on ODK-X Survey. These 2 apps, though both customized through JS, do not have the same API. I'd suggest you look into how the current `instances` UI works and also how the other prompt types are implemented. All the prompt types are in [`prompts.js`](https://github.com/odk-x/app-designer/blob/development/app/system/survey/js/prompts.js) and their template in https://github.com/odk-x/app-designer/tree/development/app/system/survey/templates. Reading the handlebars documentation might also help (https://handlebarsjs.com/).

**Comment by khushishikhu** [comment]
@linl33 yes sure, I will read the article that you have provided and implement the changes accordingly in the PR . Thank you for reviewing and guiding me it will help me to solve the issue.
